pkg/block/azure/adapter.go

```diff
@@ -190,46 +185,32 @@ func (a *Adapter) Get(ctx context.Context, obj block.ObjectPointer, _ int64) (io
 	return a.Download(ctx, obj, 0, blockblob.CountToEnd)
 }
 
-func (a *Adapter) GetPreSignedURL(ctx context.Context, obj block.ObjectPointer, mode block.PreSignMode) (string, error) {
-	permissions := aztables.AccountSASPermissions{Read: true}
+func (a *Adapter) GetPreSignedURL(_ context.Context, obj block.ObjectPointer, mode block.PreSignMode) (string, error) {
+	permissions := sas.BlobPermissions{Read: true}
 	if mode == block.PreSignModeWrite {
-		permissions = aztables.AccountSASPermissions{
-			Read:   true,
-			Add:    true,
-			Update: true,
-			Delete: true,
+		permissions = sas.BlobPermissions{
+			Read:  true,
+			Add:   true,
+			Write: true,
+			Tag:   true,
 		}
 	}
-	return a.getPreSignedURL(ctx, obj, permissions)
+	return a.getPreSignedURL(obj, permissions)
 }
 
-func (a *Adapter) getPreSignedURL(ctx context.Context, obj block.ObjectPointer, permissions aztables.AccountSASPermissions) (string, error) {
+func (a *Adapter) getPreSignedURL(obj block.ObjectPointer, permissions sas.BlobPermissions) (string, error) {
 	qualifiedKey, err := resolveBlobURLInfo(obj)
 	if err != nil {
 		return "", err
 	}
 
-	if a.keyCredentials == nil {
-		err = fmt.Errorf("pre-signed mode on Azure is only supported for shared key credentials: %w", ErrNotImplemented)
-		a.log(ctx).WithError(err).Error("no support for pre-signed using this Azure credentials provider")
-		return "", err
-	}
-
-	client, err := aztables.NewServiceClientWithSharedKey(qualifiedKey.ContainerURL, a.keyCredentials, nil)
-	if err != nil {
-		return "", err
-	}
-
-	qp, err := client.GetAccountSASURL(aztables.AccountSASResourceTypes{
-		Container: true,
-		Object:    true,
-	}, permissions, time.Now(), a.preSignedURLDurationGenerator())
+	blobURL := a.client.NewContainerClient(qualifiedKey.ContainerName).NewBlobClient(qualifiedKey.BlobURL)
+	u, err := blobURL.GetSASURL(permissions, time.Unix(0, 0), a.preSignedURLDurationGenerator())
```

Comment:
```suggestion
	u, err := blobURL.GetSASURL(permissions, time.Time{}, a.preSignedURLDurationGenerator())
```
This should also work